src/components/ListView.js

```diff
@@ -1,17 +1,36 @@
-import React from "react";
-import { useCollection } from "react-firebase-hooks/firestore";
-import { db } from "../lib/firebase";
+import React, { Fragment } from 'react';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { db } from '../lib/firebase';
+import { firestore } from 'firebase';
 
 const ListView = () => {
   const [value, loading, error] = useCollection(
-    db.collection(localStorage.getItem("token")),
+    db.collection(localStorage.getItem('token')),
     {
       snapshotListenOptions: { includeMetadataChanges: true },
     },
   );
 
   const deleteItemHandler = (id) => {
-    db.collection(localStorage.getItem("token")).doc(id).delete();
+    db.collection(localStorage.getItem('token')).doc(id).delete();
+  };
+
+  const onPurchaseChange = (e) => {
+    if (e.target.checked) {
+      db.collection(localStorage.getItem('token'))
+        .doc(e.target.value)
+        .update({ lastPurchasedDate: new Date() });
+    }
+  };
+
+  const checkIfItemHasBeenPurchased = (lastPurchaseTimestamp) => {
```

Comment:
Naming is hard. `checkIf` doesn't communicate a boolean value. A common convention for checks like this that return a boolean value is to use `is`, `has`, or `can`. In this case I would use `hasItemBeenPurchased`. Though JavaScript doesn't use it, languages like Ruby will often use a question mark to further clarify that a function returns a boolean (e.g. `hasBeenPurchased?`).

Example of why is/has/can describes a boolean behavior better:
"Honey, check if there is a mouse in the basement" ... "Ok"
"Honey, is there a mouse in the basement?" ... "Yes"

Resources:
https://www.samanthaming.com/tidbits/34-better-boolean-variable-names/
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig